.changeset/great-toys-train.md

```diff
@@ -0,0 +1,5 @@
+---
+'@astrojs/starlight': patch
+---
+
+Ensures that pages not matching the default locale slug in projects without a root locale fallback to the default locale.
```

Comment:
Feels like this could maybe be simplified somehow, gives a bit of a headache to read 🤯

Reply:
I tried!
```suggestion
Fixes a URL localization edge case. In projects without a root locale configured, slugs without a locale prefix did not fall back to the default locale as expected.
```